java/arcs/core/util/BigInt.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.util
+
+/**
+ * Multiplatform BigInteger for Java/JS/Native. Delegates to [java.math.BigInteger] on the JVM,
+ * native ES6 "BigInt" on JS, and a hand-rolled shim on native platforms.
+ */
+class BigInt private constructor(
```

Comment:
Should we have tests for this? At least for the JVM codepath?

Reply:
Josh has a similar PR, I'm letting him land his, he can steal the JS implementation from mine.

Reply:
Thanks, I've landed the PR with your BigInt impls and taking your approach for the platform independence.

Writing a showcase using them now but will add some tests that the functionality exists and is relatively sane (I don't intend to test them exhaustively).